toolkit/tools/imagegen/configuration/networkconfig_test.go

```diff
@@ -134,28 +135,12 @@ func TestShouldFailParsingInvalidDevice_Network(t *testing.T) {
 }
 
 func TestShouldPassCreatingNetworkFile_Network(t *testing.T) {
-	const (
-		networkFileDir  = "/etc/systemd/network"
-		testNetworkFile = "/etc/systemd/network/10-static-eth1.network"
-	)
-
-	// Some systems may not have systemd-networkd service configured, and thus
-	// /etc/systemd/network may not exist. For this test case, manually create this directory
-	// if it does not exist
-	exists, err := file.DirExists(networkFileDir)
-	assert.NoError(t, err)
-	if !exists {
-		err = os.Mkdir(networkFileDir, os.ModePerm)
-		assert.NoError(t, err)
-		t.Cleanup(func() {
-			err = os.RemoveAll(networkFileDir)
-			assert.NoError(t, err)
-		})
-	}
+	testNetworkFileDir := t.TempDir()
+	testNetworkFile := filepath.Join(testNetworkFileDir, "10-static-eth1.network")
 
 	testNetwork := validNetworks[0]
 
-	err = createNetworkConfigFile(nil, testNetwork, "eth1")
+	err := createNetworkConfigFile(nil, testNetwork, "eth1", testNetworkFileDir)
 	assert.NoError(t, err)
 	t.Cleanup(func() {
 		os.Remove(testNetworkFile)
```

Comment:
Do we also need to remove the tempDir (testNetworkFileDir) we created?

Reply:
No, not as far as I can tell but I will double-check. The official [docs](https://pkg.go.dev/testing#T.TempDir) say this:

>TempDir returns a temporary directory for the test to use. The directory is automatically removed by Cleanup when the test and all its subtests complete. Each subsequent call to t.TempDir returns a unique directory; if the directory creation fails, TempDir terminates the test by calling Fatal.

Reply:
I've used it before, it should clear up after the tests are done. 

Fun little tidbit that caught me off guard, I missed the bit "Each subsequent call to t.TempDir returns a unique directory" and from the top part of the description assumed it was unique to the test, but it can create multiple unique directories even inside a single test.